src/system/utils/utils.h

```diff
@@ -21,7 +21,9 @@ template<typename T, typename V, typename U> static constexpr T lmpd_constrain(c
 
 template<typename T, typename U> static constexpr U lmpd_map(T x, T in_min, T in_max, U out_min, U out_max)
 {
-  return ((float)out_max - out_min) * (((float)x - in_min) / float((float)in_max - in_min)) + out_min;
+  return static_cast<U>(
```

Comment:
* Replacing c-style cast with static_cast
* Ensuring everything needed is cast to higher precision type for range mapping calculations
* Ensuring result is cast back to output type explicitly

I put double here to avoid issues if we use this mapping on doubles.
However, if we use float-level precision at best, I can replace those doubles to float to avoid wasting resources with unneeded precision.

Reply:
Ok! Updated with floats until we need doubles.